### PR TITLE
AG-8215 Add `legendItemKey` to pie series to connect multiple pie series in the legend

### PR DESCRIPTION
**Pull request by lsjroberts**
PR for https://ag-grid.atlassian.net/browse/AG-8215

### CODE REVIEW

File: charts-community-modules/ag-charts-community/src/chart/legend.ts [hunk 1]
```diff
@@ -808,29 +808,42 @@ export class Legend {
                 (ls, s) => [...ls, ...s.getLegendData()],
                 [] as Array<LegendDatum>
             );
+
             const visibleItemsCount = legendData.filter((d) => d.enabled).length;
             const clickedItem = legendData.find((d) => d.itemId === itemId && d.seriesId === seriesId);
+
+            const seriesItemCounts = legendData.reduce((acc, d) => {
+                acc[d.seriesId] ? ++acc[d.seriesId] : (acc[d.seriesId] = 1);
```
Comment (alantreadway) [inline charts-community-modules/ag-charts-community/src/chart/legend.ts:816]: Nit: There's quite a bit going on here, could we break this down a bit to make it more readable, e.g.:
```suggestion
                acc[d.seriesId] ??= 0;
                acc[d.seriesId]++;
```

File: charts-community-modules/ag-charts-community/src/chart/legend.ts [hunk 2]
```diff
@@ -808,29 +808,42 @@ export class Legend {
...
+                return acc;
+            }, {} as any);
+            const seriesItemEnabledCounts = legendData.reduce((acc, d) => {
+                if (!d.enabled) return acc;
+                acc[d.seriesId] ? ++acc[d.seriesId] : (acc[d.seriesId] = 1);
```
Comment (alantreadway) [inline charts-community-modules/ag-charts-community/src/chart/legend.ts:821]: Likewise

**Review by alantreadway — APPROVED** [review]
LGTM other than some nits.

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-overview/examples/pie-in-a-doughnut/main.ts [hunk 3]
```diff
@@ -65,6 +65,7 @@ const options: AgChartOptions = {
       ...sharedSeriesOptions,
       data: getData2020(),
       outerRadiusRatio: 0.5,
+      legendItemKey: "browser",
```
Comment (lsjroberts) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-overview/examples/pie-in-a-doughnut/main.ts:68]: Didn't mean to commit these but I'll sort it out in a docs pr in a moment

**Pull request merged**